### PR TITLE
Move ballerina build, test and publish into a single task

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$Subject

issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/1247

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog

### CODE REVIEW

File: java.jdbc-ballerina/build.gradle [hunk 1]
```diff
@@ -265,135 +269,130 @@ task initializeVariables {
...
+                    commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat test ${testCoverageParam} ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
+                } else {
+                    commandLine 'sh', '-c', "$balJavaDebugParam $distributionBinPath/bal test ${testCoverageParam} ${groupParams} ${disableGroups} ${debugParams}"
+                }
             }
         }
-        // extract bala file to artifact cache directory
-        file("$project.projectDir/target/bala").eachFileMatch(~/.*.bala/) { balaFile ->
+
+        if (needBuildWithTest) {
```
Comment (BuddhiWathsala) [inline java.jdbc-ballerina/build.gradle:317]: Need to be else if https://github.com/ballerina-platform/module-ballerina-time/pull/100

File: java.jdbc-ballerina/build.gradle [hunk 2]
```diff
@@ -265,135 +269,130 @@ task initializeVariables {
...
+                    }
+                }
             } else {
                 throw new InvalidUserDataException("Central Access Token is Not Present")
             }
-        } else {
-            println("Publish task does not publish to ballerina repository")
+        }
+
+        if (needPublishToLocalCentral) {
```
Comment (BuddhiWathsala) [inline java.jdbc-ballerina/build.gradle:379]: Should be else if https://github.com/ballerina-platform/module-ballerina-time/pull/100

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: java.jdbc-ballerina/build.gradle (hunk 2 above)
Comment (lahirusamith) [reply java.jdbc-ballerina/build.gradle]: fixed in https://github.com/ballerina-platform/module-ballerinax-java.jdbc/pull/130/commits/2c924a74de8688cb4ef635170dc801d935323a8a

File: java.jdbc-ballerina/build.gradle (hunk 1 above)
Comment (lahirusamith) [reply java.jdbc-ballerina/build.gradle]: fixed in https://github.com/ballerina-platform/module-ballerinax-java.jdbc/pull/130/commits/2c924a74de8688cb4ef635170dc801d935323a8a

File: java.jdbc-ballerina/build.gradle [hunk 3]
```diff
@@ -502,25 +497,14 @@ copyStdlibs.dependsOn unpackStdLibs
...
 build.dependsOn ballerinaBuild
 
-ballerinaPublish.finalizedBy revertTomlFile
-ballerinaPublish.dependsOn initializeVariables
-ballerinaPublish.dependsOn updateTomlFile
-ballerinaPublish.dependsOn ":${packageName}-native:build"
-ballerinaPublish.dependsOn ballerinaBuild
-publish.dependsOn ballerinaPublish
+publishToMavenLocal.dependsOn build
+publish.dependsOn build
```
Comment (BuddhiWathsala) [inline java.jdbc-ballerina/build.gradle:510]:
```suggestion
publish.dependsOn build

```

**Review by BuddhiWathsala — APPROVED** [review]

**Comment by BuddhiWathsala** [comment]
@lahirusamith what is the behavior of this call?
```
./gradlew clean build -x test
```

Does it create DB containers unnecessarily?

**Comment by lahirusamith** [comment]
> @lahirusamith what is the behavior of this call?
> 
> ```
> ./gradlew clean build -x test
> ```
> 
> Does it create DB containers unnecessarily?

No, It does not. gradle process ignore the container creation if we mention `-x test` in the command

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**